### PR TITLE
Front end - fix app initialization

### LINKED ISSUE #60: Dashboard not displaying won blocks
Have had farmer running all day with node, no won blocks listed. Will be uploading logs here shortly. 

image

### PR DESCRIPTION
**Pull request by 1devNdogs**
# Description

Lot of changes, to transform the "fake" features to real implementation.  Also think that `client.ts` has grown a lot and will be nice to refactor in the future to improve the farmedBlock list, API-related methods, and the way it works from different pages. 

## Summary.

Launch the app for the first time. 
- Setup Plot estimate size from the network. 
  - #44 
- Plotting Progress validates Node sync and Plot status. Locally and from the network.
  - #49 
- User backed up mnemonic.
- Start farming 
- If all is OK then go to Dashboard.

Not first-time app launch.
- Validate config and Start farmer. 
- Redirect to the dashboard. 
   - #60
- Show real plot status and size. 
   - #44 
- Show real node sync and blocks.
- Load and store farmedBlocks and rewards.

User reset app from MainMenu.
- Remove and clear config file.
- Clear LocalStorage.  
- Relaunch to killing farmer.

image

### CODE REVIEW

**Comment by ozgunozerk** [comment]
Overall, great changes and thank you very much Leo!

I have 3 small requests/comments:
1. mnemonic was not displayed when I clicked on `reveal` on the info modal during plotting progress.
2. In the code, some whitespace was introduced, are you using any linter? If not, I suggest vetur. 
3. This PR is a bit heavy on the changes amount. So, I don't think *squash merge* is a good idea. We probably want to include the commits to the history for this big PR. However, commits can be re-ordered IMO. Right now, it is a bit chaotic. Rebase can be a good idea for this PR.

File: src/lib/client.ts [hunk 1]
```diff
@@ -141,15 +146,17 @@ export class Client {
   public async getNetworkSegmentIndex(hash?: Hash): Promise<number> {
     let signedBlock;
     if (hash) signedBlock = await this.publicApi.rpc.chain.getBlock(hash);
-    else signedBlock = await this.publicApi.rpc.chain.getBlock();
-    if (signedBlock.block.header.number.toNumber() === 0) return 0;
+    else signedBlock = await this.publicApi.rpc.chain.getBlock();
```
Comment (ozgunozerk) [inline src/lib/client.ts:149]: whitespace

File: src/pages/PlottingProgress.vue [hunk 2]
```diff
@@ -15,7 +15,7 @@ q-page.q-pa-lg.q-mr-lg.q-ml-lg
             outlined
             readonly
             v-model="plotDirectory"
-          )
+          )
```
Comment (ozgunozerk) [inline src/pages/PlottingProgress.vue:18]: whitespace

File: src/pages/SetupPlot.vue [hunk 3]
```diff
@@ -54,7 +54,7 @@ q-page.q-pa-lg.q-mr-lg.q-ml-lg
               )
                 q-tooltip.q-pa-sm
                   p {{ lang.availableSpace }}
-              .q-mt-sm {{ lang.allocated }}
+              .q-mt-sm {{ lang.allocated }}
```
Comment (ozgunozerk) [inline src/pages/SetupPlot.vue:57]: and here :)

**Comment by 1devNdogs** [comment]
> 1. mnemonic was not displayed when I clicked on `reveal` on the info modal during plotting progress.

This is a not covered use case, that is fixed on the `front-end-node-integration` branch, i was working with a fully synced local node so the farmer started instantly on page load, if the node is not fully synced the farmer is not started and mnemonic not available yet. As i was working on node integration i faced this issue and fixed the problem.

**Review by ozgunozerk — APPROVED** [review]

**Review by ozgunozerk — CHANGES_REQUESTED** [review]
will approve after the rebase

**Review by ozgunozerk — APPROVED** [review]
:)

**Pull request merged**